src/diffusers/models/unet_1d.py

```diff
@@ -190,6 +190,16 @@ def __init__(
             fc_dim=block_out_channels[-1] // 4,
         )
 
+    @property
+    def in_channels(self):
+        deprecate(
+            "in_channels",
+            "1.0.0",
+            "Accessing `in_channels` directly via unet.in_channels is deprecated. Please use `unet.config.in_channels` instead",
+            standard_warn=False,
+        )
+        return self.config.in_channels
+
```

Comment:
Doing this via `__getattr__` we could deprecate all properties, not just `in_channels`.

Reply:
Ah yeah good idea

Reply:
Ok played around with it a bit and don't feel super comfortable overwriting `__getattr__` actually. Many Python methods such as `hasattr(...)` rely on `__getattr__` and I don't manage to escape infinite recursions and have nice standard error messages here. 

Added deprecation properties for all the important direct accesses, think that should be good enough.